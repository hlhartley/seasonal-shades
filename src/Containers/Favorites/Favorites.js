import React, { Component } from 'react';
import { connect } from 'react-redux';
import ColorCard from '../ColorCard/ColorCard';

export class Favorites extends Component {
    render() {
        if(Object.keys(this.props.favorites).length) {
            return (
                <div>
                <h2 className='favorite-colors-text'>FAVORITE COLORS:</h2>
                <div className='favorite-colors-container'>
                {Object.keys(this.props.favorites).map((favoriteColor) => {
                    return (
                        <ColorCard color={favoriteColor}/>
                        )
                    })
                }
                </div>
                </div>
            )
        } else {
            return(
                <div className='no-favorites-container'>
                    <i class="fas fa-heart-broken"></i>
                    <p className='no-favorites-text'>No favorites found ...</p>
                </div>
            )
        }
    }
}


export const mapStateToProps = (state) => ({
    allColors: state.allColors,
    favorites: state.favorites,
});

export default connect(mapStateToProps)(Favorites);