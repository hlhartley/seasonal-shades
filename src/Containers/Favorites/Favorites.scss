.favorite-colors-container {
    align-items: center;
    display: flex;
    justify-content: center;
}

.favorite-colors-text {
    align-items: center;
    display: flex;
    justify-content: center;
}

.no-favorites-container {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin-top: 35px;
}

.no-favorites-text {
    font-size: 20px;
}

.fa-heart-broken {
    color: $grapefruit-color;
    font-size: 40px;
}