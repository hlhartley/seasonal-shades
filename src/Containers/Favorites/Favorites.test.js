import React from 'react';
import Favorites, { mapStateToProps } from './Favorites';
import { shallow } from 'enzyme';

describe('Favorites', () => {
    let wrapper;

    describe('Favorites component', () => {
        const allColors = {
            522: {
                product: {},
                hexcode: '#EED8BE'
            },
            771: {
                product: {},
                hexcode: '#BD9E9B' 
            },
            'casablanca': {
                product: {},
                hexcode: '#444446'
            }
        }
        const mockFavorites = {
            '006infinity': true,
            '002coralshape': true,
        }
        beforeEach(() => {
            wrapper = shallow(<Favorites allColors={allColors} favorites={mockFavorites}/>)
        });

        it('should match the snapshot with everything passed in correctly', () => {
            expect(wrapper).toMatchSnapshot()
        });
    });

    describe('mapStateToProps', () => {
        const mockState = {
            allColors: {
                522: {
                    product: {},
                    hexcode: '#EED8BE'
                },
                771: {
                    product: {},
                    hexcode: '#BD9E9B' 
                },
                'casablanca': {
                    product: {},
                    hexcode: '#444446'
                }
            },
            favorites: {
                '006infinity': true,
                '002coralshape': true,
            },
            blushColors: undefined,
            eyeshadowColors: undefined,
        };
        const expected = {
            allColors: {
                522: {
                    product: {},
                    hexcode: '#EED8BE'
                },
                771: {
                    product: {},
                    hexcode: '#BD9E9B' 
                },
                'casablanca': {
                    product: {},
                    hexcode: '#444446'
                }
            },
            favorites: {
                '006infinity': true,
                '002coralshape': true,
            },
        };
        const mappedProps = mapStateToProps(mockState)
        expect(mappedProps).toEqual(expected)
    });
});
