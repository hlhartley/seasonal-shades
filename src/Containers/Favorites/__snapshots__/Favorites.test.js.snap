// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites Favorites component should match the snapshot with everything passed in correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
